src/hashtree.h

```diff
@@ -23,18 +23,32 @@ SOFTWARE.
 */
 #ifndef HASHTREE_H
 #define HASHTREE_H
+
 #include <stdint.h>
 
-void hashtree_init();
+typedef void (*hashtree_hash_fcn)(unsigned char*, const unsigned char*, uint64_t);
+
+/** Perfrom hardware detection and return the fastest function available
+ *
+ * @returns Fastest hash tree function, or NULL if no supported hardware is found
+ */
+hashtree_hash_fcn hashtree_detect();
```

Comment:
what's the usage of this function? I feel like if you're willing to use whatever result this function returns then the previous behavior or `hashtree_init` is enough. If on the other hand the user may want to pass fine grained choices to override `hashtree_init`, then it's better to return here an array of options from an enum that are the supported options.

Reply:
The initial idea would be that if I have my own function pointer that I would want to fill with an implementation, I'd use this this call - however, a similar effect can be had from doing `if(!hashtree_init(NULL)) hashtree_init(fallback)`, so its utility is diminished - hidden for now, always easier to expose later than to remove: a4d60e8d7227b6e1b85446a7d471067960e464fb